examples/stable_diffusion/train_dreambooth_loha.py

```diff
@@ -0,0 +1,1103 @@
+import argparse
+import gc
+import hashlib
+import itertools
+import logging
+import math
+import os
+import threading
+import warnings
+from pathlib import Path
+from typing import Optional
+
+import datasets
+import diffusers
+import numpy as np
+import psutil
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+import transformers
+from accelerate import Accelerator
+from accelerate.logging import get_logger
+from accelerate.utils import set_seed
+from diffusers import (
+    AutoencoderKL,
+    DDPMScheduler,
+    DiffusionPipeline,
+    DPMSolverMultistepScheduler,
+    UNet2DConditionModel,
+)
+from diffusers.optimization import get_scheduler
+from diffusers.utils import check_min_version
+from diffusers.utils.import_utils import is_xformers_available
+from huggingface_hub import HfFolder, Repository, whoami
+from PIL import Image
+from torch.utils.data import Dataset
+from torchvision import transforms
+from tqdm.auto import tqdm
+from transformers import AutoTokenizer, PretrainedConfig
+
+from peft import LoHaConfig, get_peft_model
+
+
+# Will error if the minimal version of diffusers is not installed. Remove at your own risks.
+check_min_version("0.10.0.dev0")
+
+logger = get_logger(__name__)
+
+UNET_TARGET_MODULES = [
```

Comment:
Why can more layers be targeted here vs in the LoRA dreambooth script? Could we add them to LoRA as well?

Reply:
All these layers could also be incorporated into LoRA dreambooth script.

AFAIK other SD adapter training frameworks like kohya has a special flag that enables/disables Conv2d layers adapters. So it's a matter of preference, we could also incorporate a similar flag here.

Please let me know if you would like me to do it.

Reply:
Thanks for explaining. I have no strong opinion either way, I'll leave it up to you.